### ISSUE TITLE
Earth Slimestaff does not appear to work with Fire Protection modifier

### ISSUE THREAD

**Issue by zingewill** [SlimeKnights/TinkersConstruct#5169]
### Minecraft Version

1.18.2

### Forge Version

40.2.10

### Mantle Version

1.9.45

### Tinkers' Construct Version

3.6.4.113

### Describe your issue

I applied Fire Prot 4 to an Earthsline Staff (using rebalanced modifier to gain the fourth defense slot), and after several tests, concluded that it simply does not work.
Tests were done with staff in main, then offhand, first with enchanted vanilla/modded armor, then with no armor, then wearing unmodified Plate armor. The unarmored test was reproduced in a singleplayer Creative world, then again on a similar world in a modpack with only Mantle and Tinkers'. Test was performed simply by walking into lava then jumping out, noting damage taken and number of fire ticks. No additional reduction in damage was noted, fire ticks remained at 15, and no durability was lost on staff in any of the tests. No tests were done with other kinds of protection modifiers added to it, nor were tests done with other varieties of Slimestaff.

### Other mods

Mantle, Tinkers'

### Tried reproducing with just Tinkers?

Yes

### Performance Enchancers

None of the above

### Searched for known issues?

Searched open issues

**Comment by KnightMiner** [comment]
Please do more tests to narrow down the issue. Does the same thing happen with shields with the modifier? What about a single piece of tinkers armor with the modifier?

Its possible that the current implementation of fire protection means fire tick reduction from shields just will not work, sideeffect of using the vanilla logic. But you should still see reduction in damage from fire from shields.

**Comment by zingewill** [comment]
I tested it with Plate armor and Plate shield, on the clean instance. Fire prot works exactly as intended on Plate armor, but has no effect on Plate shield, just like the slimestaff. I also tested Blast, Melee, Magic, and Projectile on Plate shield; all four of these work as stated (though for the additional effect, I only confirmed KB resist from blast prot works, the rest are assumed to work). 
Noted that the 2 hearts of damage from the lava itself is not reduced in any way either by the fire prot plate shield, which was also observed with slimestaff, but was reduced by the plate armor to 1/2 to 1 heart. All of these tests were done with level 5 of each modifier.
